drools-wb-screens/drools-wb-globals-editor/drools-wb-globals-editor-client/src/main/java/org/drools/workbench/screens/globals/client/editor/GlobalsEditorViewImpl.java

```diff
@@ -108,66 +113,64 @@ public String getValue( final Global global ) {
             }
         };
 
-        final ButtonCell deleteGlobalButton = new ButtonCell( IconType.MINUS, ButtonType.DANGER, ButtonSize.SMALL );
+        deleteGlobalButton = new ButtonCell( IconType.MINUS,
+                                             ButtonType.DANGER,
+                                             ButtonSize.SMALL );
         final Column<Global, String> deleteGlobalColumn = new Column<Global, String>( deleteGlobalButton ) {
             @Override
             public String getValue( final Global global ) {
-                return GlobalsEditorConstants.INSTANCE.remove();
+                return translationService.getTranslation( GlobalsEditorConstants.GlobalsEditorViewImplRemove );
             }
         };
-        deleteGlobalColumn.setFieldUpdater( new FieldUpdater<Global, String>() {
-            public void update( final int index,
-                                final Global global,
-                                final String value ) {
-                if ( isReadOnly ) {
-                    return;
-                }
-                if ( Window.confirm( GlobalsEditorConstants.INSTANCE.promptForRemovalOfGlobal0( global.getAlias() ) ) ) {
-                    dataProvider.getList().remove( index );
-                }
+        deleteGlobalColumn.setFieldUpdater( ( index, global, value ) -> {
+            if ( Window.confirm( translationService.format( GlobalsEditorConstants.GlobalsEditorViewImplPromptForRemovalOfGlobal,
```

Comment:
Surely, if the global was generated for OptaPlanner, you'd want to prevent removal from the table too? I check against deletion of the file containing the generated global but not against removal of the definition from the file itself. Have I missed something?